plancheck/expect_sensitive_value_test.go

```diff
@@ -0,0 +1,274 @@
+				return testProviderSensitive(), nil
+			},
+		},
+		Steps: []r.TestStep{
+			{
+				Config: `
+				resource "test_resource" "one" {
+					sensitive_map_attribute = {
+						key1 = "value1",
+					}
+				}
+				`,
+				ConfigPlanChecks: r.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectSensitiveValue("test_resource.one",
+							tfjsonpath.New("sensitive_map_attribute")),
+					},
+				},
+			},
+		},
+	})
+}
+
+func Test_ExpectSensitiveValue_ListNestedBlock_SensitiveAttribute(t *testing.T) {
+	t.Parallel()
+
+	r.UnitTest(t, r.TestCase{
+		ProviderFactories: map[string]func() (*schema.Provider, error){
+			"test": func() (*schema.Provider, error) { //nolint:unparam // required signature
+				return testProviderSensitive(), nil
+			},
+		},
+		Steps: []r.TestStep{
+			{
+				Config: `
+				resource "test_resource" "one" {
+					list_nested_block_sensitive_attribute {
+						sensitive_list_nested_block_attribute = "sensitive-test"
+						list_nested_block_attribute = "test"
+					}
+				}
+				`,
+				ConfigPlanChecks: r.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectSensitiveValue("test_resource.one",
+							tfjsonpath.New("list_nested_block_sensitive_attribute").AtSliceIndex(0).
+								AtMapKey("sensitive_list_nested_block_attribute")),
+					},
+				},
+			},
+		},
+	})
+}
+
+func Test_ExpectSensitiveValue_SetNestedBlock_SensitiveAttribute(t *testing.T) {
+	t.Parallel()
+
+	r.UnitTest(t, r.TestCase{
+		ProviderFactories: map[string]func() (*schema.Provider, error){
+			"test": func() (*schema.Provider, error) { //nolint:unparam // required signature
+				return testProviderSensitive(), nil
+			},
+		},
+		Steps: []r.TestStep{
+			{
+				Config: `
+				resource "test_resource" "one" {
+					set_nested_block_sensitive_attribute {
+						sensitive_set_nested_block_attribute = "sensitive-test"
+						set_nested_block_attribute = "test"
+					}
+				}
+				`,
+				ConfigPlanChecks: r.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectSensitiveValue("test_resource.one",
+							tfjsonpath.New("set_nested_block_sensitive_attribute")),
+					},
+				},
+			},
+		},
+	})
+}
+
+func testProviderSensitive() *schema.Provider {
+	return &schema.Provider{
+		ResourcesMap: map[string]*schema.Resource{
+			"test_resource": {
+				CreateContext: func(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
+					d.SetId("test")
+					return nil
+				},
+				UpdateContext: func(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+					return nil
+				},
+				DeleteContext: func(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+					return nil
+				},
+				ReadContext: func(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+					return nil
+				},
+				Schema: map[string]*schema.Schema{
+					"sensitive_string_attribute": {
```

Comment:
Should we consider adding a test that uses `ProtoV6ProviderFactories` to cover list, map and set nested attributes?

Reply:
Due to the current limitations of the testing module, we cannot create Framework providers for testing within this repo without adding a dependency to `terraform-plugin-framework.` Because of this, we cannot test any protocol v6 types since the SDKv2 doesn't support protocol v6. We can consider writing these tests in a separate repo, if we feel that there's value in that.

Reply:
Fair. I have wondered previously about whether or not it would be ok to add a dependency on `terraform-plugin-framework` for this purpose.